use contents of startup_script_1 and startup_script_2.

bmenasha opened the pull request:
previously the examples would fail as the startup-script metadata was the string
"../../modules/sap_hana/files/startup.sh" rather then the actual startup script

Tfmenard commented on modules/sap_hana_ha/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+ * limitations under the License.
+ */
+
+output "instance_primary_address" {
```
Should be `primary_instance_name`
The rest of the output should be consistent as well like this.
`primary_instance_*`

arvindpai replied:
it needs to be master and worker as per SAP notes on their website

arvindpai replied:
made those changes .

Tfmenard commented on modules/sap_hana_ha/outputs.tf:
```diff
@@ -0,0 +1,47 @@
+  value = "${google_compute_instance.primary.machine_type}"
+}
+
+output "instance_secondary_address" {
```
Same here.
Should be secondary_instance_name
The rest of the output should be consistent as well like this.
secondary_instance_*

arvindpai replied:
done with these changes

Tfmenard commented on examples/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,155 @@
+  description = "List of network tags"
+}
+
+variable "gcp_primary_instance_ip" {
```
Remove the `gcp` prefixes from all the variables with this prefix if this is not in the DM templates variables.

arvindpai replied:
changes completed.

Tfmenard commented on modules/sap_hana_ha/main.tf:
```diff
@@ -0,0 +1,208 @@
+
+    # Needed for startup-scripts module
+    startup-script        = "${var.startup_script_2}"
+    startup_script_custom = "${var.startup_script_custom}"
```
Remove startup_script_custom variablle and any reference to it. Also remove the comment as this is not true anymore.

arvindpai replied:
this has been removed.

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,155 @@
+ */
+# Changed from SCR
+
+variable "sap_primary_instance" {
```
Remove the `sap` prefix and chhange this variable to `primary_instance_name`.
Using the `sap` prefix is redundant.

To this for all variables thathave sap prefix and that are not present in the DM template.
https://cloud.google.com/solutions/sap/docs/sap-hana-ha-deployment-guide#creating_a_high-availability_linux_cluster_with_sap_hana_installed

arvindpai replied:
removed

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,155 @@
+  description = "List of network tags"
+}
+
+variable "gcp_primary_instance_ip" {
```
Remove `gcp` prefix from the variable name and from the description.

arvindpai replied:
removed

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,155 @@
+}
+
+# TODO: Delete if not necessary
+variable "startup_script_custom" {
```
Remove this variable it is no longer needed

arvindpai replied:
removed

Tfmenard commented on test/fixtures/sap_hana_ha_simple_example/main.tf:
```diff
@@ -0,0 +1,95 @@
+locals {
+  gcs_bucket_name = "post-deployment-bucket-${random_id.random_suffix.hex}"
+
+  # TODO: Add requirements of downloading sotware in README.md of module
```
Remove this comment

arvindpai replied:
Removed

Tfmenard commented on test/fixtures/sap_hana_ha_simple_example/outputs.tf:
```diff
@@ -14,12 +14,32 @@
   value       = "${var.project_id}"
 }
+
+output "sap_primary_instance" {
```
Remove the `sap` prefix from all these output names and make their name and description consistent with changes made in the examples.

arvindpai replied:
sap prefix removed from outputs.tf

Tfmenard commented on test/fixtures/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,141 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+# Changed from SCR
```
remove this comment

arvindpai replied:
comment removed.

Tfmenard commented on test/fixtures/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,141 @@
+ */
+# Changed from SCR
+
+variable "sap_primary_instance" {
```
Same here. Make the same changes you made in the examples folder and in the modules folder to variables name and decription,

arvindpai replied:
removed that comment

Tfmenard commented on test/integration/sap_hana_ha_simple_example/inspec.yml:
```diff
@@ -0,0 +1,30 @@
+  - name: bucket_name
+    required: true
+    type: string
+  - name: sap_primary_instance
```
Don't forget to update the variables here as well after renaming the outputs in test/fixtures/sap_hana_ha_simple_example/outputs.tf

arvindpai replied:
sap prefix removed from the variables.tf and outputs.tf

arvindpai commented on examples/sap_hana_ha_simple_example/main.tf:
```diff
@@ -0,0 +1,62 @@
+  version = "~> 1.18.0"
+
+  # TODO: Take out project if not needed
+  project = "${var.project_id}"
```
Made those changes in the code

Tfmenard commented on modules/sap_hana_ha/main.tf:
```diff
@@ -0,0 +1,208 @@
+  required_version = "~> 0.11.0"
+}
+
+resource "google_compute_address" "gcp-primary-instance-ip" {
```
All resources name should have underscores not hyphens.
gcp-primary-instance-ip should be gcp_primary_instance_ip

arvindpai replied:
made the stated changes in the code

arvindpai commented on modules/sap_hana_ha/files/startup_secondary.sh:
```diff
@@ -0,0 +1,13 @@
+chmod +x startup_secondary.sh
+source startup_secondary.sh
+
+# Verifying HANA HA deployment
```
removed that comment

Tfmenard commented on examples/sap_hana_ha_simple_example/main.tf:
```diff
@@ -0,0 +1,58 @@
+ */
+
+provider "google" {
+  version = "~> 1.18.0"
```
Update to 2.6.0
SAP HANA submodule and test again.
This update has been done for SAP HAN submodule

arvindpai replied:
made those changes to the code.

arvindpai replied:
done

Tfmenard commented on examples/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+# Changed from SCR
```
Remove this comment

arvindpai replied:
removed

arvindpai replied:
done

Tfmenard commented on modules/sap_hana_ha/README.md:
````diff
@@ -0,0 +1,128 @@
+You can go in the [examples](../../examples) folder complete working example. However, here's an example of how to use the module in a main.tf file.
+
+```hcl
+provider "google" {
````
Changed provider version here as well.

arvindpai replied:
done

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+# Changed from SCR
```
Remove this comment

arvindpai replied:
done

arvindpai commented on examples/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,155 @@
+}
+
+# TODO: Delete if not necessary
+variable "startup_script_custom" {
```
removed it from the code wherever it existed.

arvindpai replied:
removed

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+# Changed from SCR
```
Remove this comment

arvindpai replied:
removed that comment

arvindpai commented on examples/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,155 @@
+ */
+# Changed from SCR
+
+variable "sap_primary_instance" {
```
removed

arvindpai replied:
done

arvindpai commented on modules/sap_hana_ha/files/startup.sh:
```diff
@@ -0,0 +1,13 @@
+chmod +x startup.sh
+source startup.sh
+
+# Verifying HANA HA deployment
```
done

arvindpai replied:
done

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,149 @@
+ */
+# Changed from SCR
+
+variable "primary_instance" {
```
You'll need to update variables and their descriptions that appear in SAP HANA submodule as well to stay consistent.
See the modules/sap_hana/variables.tf file here for reference.
https://github.com/terraform-google-modules/terraform-google-sap/pull/2/files

arvindpai replied:
made those changes

Tfmenard replied:
These changes weren't completed.
For example, project_id's description is: ' description = "The ID of the project in which the resources will be deployed." '

For instance_name variables in SAP HANA the description is:
'description = "A unique name for the resource, required by GCE. Changing this forces a new resource to be created."'

Please review the SAP HANA variables description and name and make the changes in here for consistency.

arvindpai replied:
done with these changes

Tfmenard commented on test/fixtures/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,140 @@
+ * limitations under the License.
+ */
+
+variable "primary_instance" {
```
After updating variables in the root module, make sure you update them here as well.

arvindpai replied:
made those changes

Tfmenard commented on examples/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,149 @@
+ */
+# Changed from SCR
+
+variable "primary_instance" {
```
After updating the root module, make sure to update all variables and description here as well.

arvindpai replied:
done

arvindpai replied:
done with these changes

Tfmenard commented on examples/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,149 @@
+  description = "Compute Engine instance name"
+}
+
+variable "primary_zone" {
```
Most variables in examples should be using defaults.
Please add defaults to all variables that are not user specific.

arvindpai replied:
I checked with Ajay, he said not to use default values anywhere

arvindpai replied:
added terraform.tfvars.example file and made appropriate changes to the README.MD file to reflect those changes.

Tfmenard commented:
Because our examples need to be tested with multiple configurations we cannot create a working an example with too much hardcoded values.
However, we still need to guide the user into a quick setup using our example.
To do that, please add a terraform.tfvars.example and add an explanation on how to use in the example's README similar to what's been done here.
https://github.com/terraform-google-modules/terraform-google-sap/pull/2/files

Tfmenard commented on examples/sap_hana_ha_simple_example/README.md:
```diff
@@ -0,0 +1,94 @@
+- `terraform destroy` to destroy the built infrastructure
+
+
+## Integration Tests
```
This ection has a little bit in SAP HANA submodule please apply those changes to be consistent with the examples/sap_hana_simple_example/README.md file for the SAP HANA submodule.

arvindpai replied:
made those corrections. Did not put any default values.

Tfmenard replied:
Please add the defaults. Example should have the least amount of configurations to be done by a user in order to quickly get started.

arvindpai replied:
I have added the default values

Tfmenard commented on examples/sap_hana_ha_simple_example/main.tf:
```diff
@@ -0,0 +1,58 @@
+  network_tags               = "${var.network_tags}"
+  project_id                 = "${var.project_id}"
+  region                     = "${var.region}"
+  service_account            = "${var.service_account}"
```
Chane this variable name to `service_account_email`

arvindpai replied:
done with those changes.

Tfmenard commented on examples/sap_hana_ha_simple_example/outputs.tf:
```diff
@@ -14,19 +14,17 @@
-  length    = 1
-  prefix    = "simple-example"
-  separator = "-"
+output "primary_instance" {
```
This should be named `primary_instance_name`. Please change it for every matching variables and outputs.

arvindpai replied:
done with these changes

Tfmenard commented on examples/sap_hana_ha_simple_example/variables.tf:
```diff
@@ -0,0 +1,149 @@
+  description = "Project id to deploy the resources"
+}
+
+variable "region" {
```
This variable should have a default value as well.
Other variables can have a default value.
Look at the examples/sap_hana_simple_example/variables.tf file for the SAP HANA module for reference

arvindpai replied:
Kindly confirm with Ajay on this . He told me to give all values in terraform.tfvars. nothing to stated as default in any variable .

Tfmenard replied:
See this file to set the appropriate default values and make it consistent with the SAP HANA submodule.
https://github.com/terraform-google-modules/terraform-google-sap/blob/master/examples/sap_hana_simple_example/main.tf

arvindpai replied:
made all those necessary changes.

Tfmenard commented on README.md:
````diff
@@ -1,24 +1,14 @@
-}
-```
+Each submodules have their own usage documented in the [modules](./modules) folder.
+For example, see the [SAP HANA HA Usage Section](./modules/sap_hana_ha/README.md#Usage).
````
Change it back to point to SAP HANA.
This root README is the starting point for users to use all modules. We should point to the simplest one, SAP HANA not SAP HANA HA.
```suggestion
For example, see the [SAP HANA Usage Section](./modules/sap_hana/README.md#Usage).
```

arvindpai replied:
Done with those changes.

Tfmenard commented on examples/sap_hana_ha_simple_example/main.tf:
```diff
@@ -0,0 +1,58 @@
+  region  = "${var.region}"
+}
+
+module "gcp_sap_hana_ha" {
```
See this file https://github.com/terraform-google-modules/terraform-google-sap/blob/master/examples/sap_hana_simple_example/main.tf to make it consistent in terms of which variables are hardcoded in the example.

arvindpai replied:
Added default as well as hard coded values in the code.

Tfmenard commented on modules/sap_hana_ha/README.md:
```diff
@@ -0,0 +1,135 @@
+  region  = "${var.region}"
+}
+
+module "gcp_sap_hana_ha" {
```
Once changes are done in the example for hardcoded values, update the Usage section here with an up to date code sample

arvindpai replied:
Updated the Usage section with the code changes inclusive of hardcoded values.

Tfmenard commented on modules/sap_hana_ha/main.tf:
```diff
@@ -0,0 +1,206 @@
+  subnetwork   = "${var.subnetwork}"
+  address_type = "INTERNAL"
+  address      = "${var.sap_vip}"
+  region       = "us-central1"
```
The region should not be hardcoded here.

Tfmenard commented on modules/sap_hana_ha/main.tf:
```diff
@@ -0,0 +1,206 @@
+  size    = "${var.pd_ssd_size}"
+}
+
+resource "google_compute_disk" "pd_standard_primary" {
```
Is a standard Standard PD and PD SSD always required for both instances of a SAP HANA HA deployment  ?

Tfmenard commented on modules/sap_hana_ha/main.tf:
```diff
@@ -0,0 +1,206 @@
+    sap_vip                    = "${var.sap_vip}"
+    sap_vip_secondary_range    = "${var.sap_vip_secondary_range}"
+
+    # Needed for startup-scripts module
```
Remove this comment

Tfmenard commented on modules/sap_hana_ha/main.tf:
```diff
@@ -0,0 +1,206 @@
+    sap_vip                    = "${var.sap_vip}"
+    sap_vip_secondary_range    = "${var.sap_vip_secondary_range}"
+
+    # Needed for startup-scripts module
```
Remove this comment

arvindpai replied:
Removed that comment.

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+
+variable "primary_instance_name" {
```
See this file to add default where possible and make it consistent with SAP HANA submodule.
https://github.com/terraform-google-modules/terraform-google-sap/blob/master/modules/sap_hana/variables.tf

arvindpai replied:
made it consistent with SAP HANA module

Tfmenard commented on test/fixtures/sap_hana_ha_simple_example/files/templates/post_deployment_script.tpl:
```diff
@@ -0,0 +1,8 @@
+#! /bin/bash
+#
```
Remove unncessary comments

arvindpai replied:
removed unnecessary comments from tpl file

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,148 @@
+}
+
+variable "primary_instance_ip" {
+  description = "gcp primary instance ip address"
```
```suggestion
  description = "Primary instance IP address"
```

arvindpai replied:
Done with these changes.

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,148 @@
+}
+
+variable "secondary_instance_ip" {
+  description = "gcp secondary instance ip address"
```
```suggestion
  description = "Secondary instance IP address"
```

arvindpai replied:
Done with these changes

Tfmenard commented on modules/sap_hana_ha/variables.tf:
```diff
@@ -0,0 +1,148 @@
+  description = "Name of static IP adress to add to the instance's access config."
+}
+
+variable "startup_script_1" {
```
I saw that you made modification of these variables in SAP Scaleout.
https://github.com/terraform-google-modules/terraform-google-sap/pull/4/commits/fb44be66515ec4add31a0b072f49df933f68bc1d
Please make it consistent between the two by choosing between startup_script_1 and startup_script_primary and make sure to be consistent between any other common variables between SAP HANA HA and SAP HANA Scaleout and while also maintaining the standards set by the SAP HANA submodule

arvindpai replied:
will make changes to scaleout module to map to these values

Tfmenard commented on examples/sap_hana_ha_simple_example/terraform.tfvars.example:
```diff
@@ -0,0 +1,85 @@
+
+pd_standard_size = 350
+
+service_account = "<gcp service account>"
```
This won't work as the variable has been change to service_account_email.
Please make sure this terraform.tfvars.example reflects your latest working terraform.tfvars file with all latest variable names and possibly updated values.

arvindpai replied:
Made the necessary changes

Tfmenard commented on modules/sap_hana_ha/README.md:
```diff
@@ -0,0 +1,147 @@
+ Follow instructions [here](https://cloud.google.com/solutions/sap/docs/sap-hana-ha-deployment-guide#creating_a_cloud_storage_bucket_for_the_sap_hana_ha_installation_files) to properly Download SAP HANA from the SAP Marketplace, and upload it to a GCS bucket.
+
+
+### Configure Service Account for identifying the Compute instance
```
```suggestion
### Configure Service Account for identifying the Compute instances
```

arvindpai replied:
Made the necessary change.

Tfmenard commented on modules/sap_hana_ha/README.md:
```diff
@@ -0,0 +1,147 @@
+
+
+### Configure Service Account for identifying the Compute instance
+The compute instance created by this submodule will need to download SAP HANA from a GCS bucket in order install it. Follow the instructions below to ensure a successful installation:
```
```suggestion
The compute instances created by this submodule will need to download SAP HANA from a GCS bucket in order install it. Follow the instructions below to ensure a successful installation:
```

arvindpai replied:
Made the necessary change.

bmenasha committed:
use contents of startup_script_1 and startup_script_2.
previously the examples would fail as the startup-script metadata was the string
"../../modules/sap_hana/files/startup.sh" rather then the actual startup script

supply startup file contents in examples

morgante commented on modules/sap_hana/main.tf:
```diff
@@ -96,7 +96,7 @@ resource "google_compute_instance" "gcp_sap_hana" {
     sap_hana_sapsys_gid        = "${var.sap_hana_sapsys_gid}"
 
-    startup-script = "${var.startup_script}"
+    startup-script = "${file(var.startup_script)}"
```
Instead of breaking the module interface, could we move this into the example code? So the module itself still accepts the actual startup script (not a file path)?

bmenasha committed:
use module's startup script and not a copy in text fixtures.

bmenasha commented:
sure. done.
I also did some cleanup on the test fixtures, no need to duplicate the startup script for each test.
thanks
